src/Illuminate/Queue/Console/FlushBatchCommand.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Illuminate\Queue\Console;
+
+use Carbon\Carbon;
+use Illuminate\Bus\BatchRepository;
+use Illuminate\Console\Command;
+
+class FlushBatchCommand extends Command
+{
+    /**
+     * The console command signature.
+     *
+     * @var string
+     */
+    protected $signature = 'queue:flush-batch {--hours=24 : The number of hours to retain batch data}';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Prune stale entries from the batches database';
+
+    /**
+     * Execute the console command.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        $count = 0;
+
+        $repository = $this->laravel[BatchRepository::class];
+
+        if (method_exists($repository, 'prune')) {
```

Comment:
I'd prefer a `Prunable` interface, or something.

Reply:
@driesvints This was changed from your previous comment can we add new contracts in a stable release?

@GrahamCampbell for reference: https://github.com/laravel/framework/pull/35694#discussion_r547353718

Reply:
New contracts are possible yes.

Reply:
Changed to use `Prunable` interface as dcussed